app/src/main/java/com/edu/news/data/source/local/CategoryLocalDataSource.kt

```diff
@@ -0,0 +1,22 @@
+package com.edu.news.data.source.local
+
+import android.content.Context
+import com.edu.news.data.model.Category
+import com.edu.news.data.source.CategoryDataSource
+import com.edu.news.data.source.local.interactData.LocalDataBaseHelper
+
+class CategoryLocalDataSource : CategoryDataSource.Local {
+
+    override fun getCategoryFromLocal(listener: OnFetchDataListener<MutableList<Category>>, context: Context) {
+        LocalDataBaseHelper(listener, context).readCategory()
+    }
+
+    override fun postCategoryToLocal(context: Context, array: Array<String>) {
+        val listener: OnFetchDataListener<MutableList<Category>>? = null
+        LocalDataBaseHelper(listener, context).addAllCategory(array)
+    }
```

Comment:
sao lại tạo listenter ở đây truyền vì mục đích gì e?
```suggestion
    override fun postCategoryToLocal(categorites: List<String>) {
        val listener: OnFetchDataListener<MutableList<Category>>? = null
        LocalDataBaseHelper(listener).addAllCategory(array)
    }
```